rest/taskana-rest-spring/src/test/java/pro/taskana/task/rest/assembler/AttachmentRepresentationModelAssemblerTest.java

```diff
@@ -0,0 +1,135 @@
+    summary.setKey("keyabc");
+    summary.setDomain("DOMAIN_A");
+    summary.setType("MANUAL");
+    AttachmentRepresentationModel repModel = new AttachmentRepresentationModel();
+    repModel.setCustomAttributes(Collections.singletonMap("abc", "def"));
+    repModel.setClassificationSummary(summary);
+    repModel.setAttachmentId("id");
+    repModel.setTaskId("taskId");
+    repModel.setChannel("channel");
+    repModel.setCreated(Instant.parse("2019-09-13T08:44:17.588Z"));
+    repModel.setModified(Instant.parse("2019-09-13T08:44:17.588Z"));
+    repModel.setObjectReference(reference);
+    repModel.setReceived(Instant.parse("2019-09-13T08:44:17.588Z"));
+
+    AttachmentImpl attachment = assembler.toEntityModel(repModel);
+
+    testEqualityAfterConversion(repModel, attachment);
+  }
+
+
+  @Test
+  void should_ReturnRepresentationModel_When_ConvertingEntityToRepresentationModel() {
+    AttachmentImpl attachment = (AttachmentImpl) taskService.newAttachment();
+    ObjectReference reference = new ObjectReference();
+    ClassificationSummary summary = classService.newClassification("ckey", "cdomain", "MANUAL");
+    reference.setId("abc");
+    attachment.setCustomAttributes(Collections.singletonMap("abc", "def"));
+    attachment.setClassificationSummary(summary);
+    attachment.setId("id");
+    attachment.setTaskId("taskId");
+    attachment.setChannel("channel");
+    attachment.setCreated(Instant.parse("2019-09-13T08:44:17.588Z"));
+    attachment.setModified(Instant.parse("2019-09-13T08:44:17.588Z"));
+    attachment.setObjectReference(reference);
+    attachment.setReceived(Instant.parse("2019-09-13T08:44:17.588Z"));
+
+    AttachmentRepresentationModel repModel = assembler.toModel(attachment);
+
+    testEqualityAfterConversion(repModel, attachment);
+    testLinks();
+  }
+
+  @Test
+  void should_Equal_When_ComparingEntityWithConvertedEntity() {
+    AttachmentImpl attachment = (AttachmentImpl) taskService.newAttachment();
+    ObjectReference reference = new ObjectReference();
+    ClassificationSummary summary = classService.newClassification("ckey", "cdomain", "MANUAL");
+    reference.setId("abc");
+    attachment.setCustomAttributes(Collections.singletonMap("abc", "def"));
+    attachment.setClassificationSummary(summary);
+    attachment.setId("id");
+    attachment.setTaskId("taskId");
+    attachment.setChannel("channel");
+    attachment.setCreated(Instant.parse("2019-09-13T08:44:17.588Z"));
+    attachment.setModified(Instant.parse("2019-09-13T08:44:17.588Z"));
+    attachment.setObjectReference(reference);
+    attachment.setReceived(Instant.parse("2019-09-13T08:44:17.588Z"));
+
+    AttachmentRepresentationModel repModel = assembler.toModel(attachment);
+    AttachmentImpl attachment2 = assembler.toEntityModel(repModel);
+
+    testEqualityOfEntities(attachment, attachment2);
+  }
+
+  void testEqualityAfterConversion(AttachmentRepresentationModel repModel, Attachment attachment) {
```

Comment:
This is not consistent. The tests above have the entity first, then the repModel